src/core/common/settings.cpp

```diff
@@ -395,6 +402,54 @@ void Settings::ChildInfoIterator::Read(void)
     mIsDone = (error != OT_ERROR_NONE);
 }
 
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+otError Settings::ReadDadInfo(DadInfo &aDadInfo) const
+{
+    otError  error;
+    uint16_t length = sizeof(DadInfo);
+
+    aDadInfo.Init();
+    SuccessOrExit(error = Read(kKeyDadInfo, &aDadInfo, length));
+    LogDadInfo("Read", aDadInfo);
+
+exit:
+    return error;
```

Comment:
no `LogFailure` for Read Op in existing settings.
Guess there is no information stored for a fresh device, so `LogFailure` is skipped for Read as may not care so much as the other operations.

Reply:
Should call `ReadDadInfo` whenever node is booted? 

Noted that other `ReadXxxxInfo` methods do not have `LogFailure` either, not knowing why. @abtink

Reply:
It can be added but as @librasungirl mentioned it does not add much new info/value to log error on read operation. 

When reading a setting entry, it's legitimate to not have the setting entry and thus returning `OT_ERROR_NOT_FOUND` (which is not really an error and more an indication that such entry does not exist). We need to handle this error as a special case in the logs since it is not useful to log such error (it'd just bloat the logs).